index.html

```diff
@@ -235,24 +236,27 @@ <h2>
           </pre>
         </li>
         <li>
-          Let |permission| be the result of running
+          Let |resultPromise| be the result of running
           <a>query a permission</a> with |permissionDesc|.
         </li>
         <li>
-          If |permission| is "<a>denied</a>",
-          return <code>false</code>.
+          Let |status:PermissionStatus| be the result of waiting for
+          |resultPromise| to settle.
         </li>
         <li>
-          If |permission| is "<a>granted</a>",
-          return <code>true</code>.
+          If |resultPromise| rejects, return "<a>denied</a>".
         </li>
         <li>
-          If |permission| is "<a>prompt</a>", run
-          the following steps:
+          Let |state| be the value of the
+          <a data-cite="permissions#dom-permissionstatus-state">state</a>
+          attribute of |status|.
+        </li>
+        <li>
+          If |state| is "<a>prompt</a>", run the following steps:
```

Comment:
It doesn't feel right for this spec to override the results of the permission grant of the permission spec. Why not let it prompt?  

Something is either wrong in this spec or in the Permissions spec.

Reply:
Permission spec allows you to query, and to prompt separately. These are separate algorithms. 

I based this on existing spec text, but Web USB does similar (though it uses "ask") as well as device motion uses "default" the same way "prompt" is used here (as that spec doesn't depend on permission api)